receiver/dockerstatsreceiver/receiver.go

```diff
@@ -126,8 +138,23 @@ func (r *Receiver) Run() error {
 	wg := &sync.WaitGroup{}
 	wg.Add(len(containers))
 	for _, container := range containers {
-		go func(dc DockerContainer) {
-			md, err := r.client.FetchContainerStatsAndConvertToMetrics(ctx, dc)
+		go func(c docker.Container) {
+			statsJSON, err := r.client.FetchContainerStatsAsJSON(ctx, c)
+			if err != nil {
+				results <- result{pdata.NewMetrics(), err}
+				wg.Done()
```

Comment:
Nit: instead of this and line 159 put `defer wg.Done()` at the top of this anonymous func. It ensures there are no paths where we forget to do it and is one line instead of two.

Reply:
I like this better too, thanks!